app/graphql/albums/resolver.js

```diff
@@ -0,0 +1,24 @@
+const { allAlbums, albumById, photosByAlbumId } = require('../../services/album');
+
+exports.getAlbum = async (_, params) => {
+  const album = await albumById(params.id);
+  const data = {
+    ...album,
+    artist: album.userId,
+    name: album.title,
+    photos: await photosByAlbumId(params.id)
+  };
+  return data;
+};
+
+exports.getAlbums = async (_, params) => {
+  const tempAlbums = await allAlbums(params);
+  const promisesPhotos = tempAlbums.map(album => photosByAlbumId(album.id));
+  const photosRensponse = await Promise.all(promisesPhotos);
+  const albums = tempAlbums.map((album, index) => ({
+    ...album,
+    artist: album.userId,
+    photos: photosRensponse[index]
+  }));
+  return albums;
```

Comment:
Hi Jonny, there is something else you can do to improve this code.

You can use type resolvers in order to only request album photos when query specify them. For more information, please visit https://www.apollographql.com/docs/graphql-tools/resolvers/ and look how it request for posts.

Please, open a new PR to make this improvement